pyomo/core/tests/unit/test_expr_pyomo5.py

```diff
@@ -6379,6 +6379,355 @@ def test_replace_nested(self):
         self.assertEqual(f.arg(0).nargs(), 4)
 
 
+class TestGenericExpressionVisitor(unittest.TestCase):
+    def setUp(self):
+        self.m = m = ConcreteModel()
+        m.x = Var()
+        m.y = Var()
+        m.z = Var()
+        self.e = m.x**2 + m.y + m.z*(m.x+m.y)
+
+    def test_bad_args(self):
+        with self.assertRaisesRegexp(
+                RuntimeError, "Unrecognized keyword arguments: {'foo': None}"):
+            EXPR.GenericExpressionVisitor(foo=None)
+
+    def test_default(self):
+        walker = EXPR.GenericExpressionVisitor()
+        ans = walker.walk_expression(self.e)
+        ref = [
```

Comment:
Hmmm... to make this more understandable, how about
```
ref = [
            [
               [],
               []
            ],
            [],
            [
               [],
               [
                  [],
                  []
               ]
            ]
        ]
```
(I am not convinced this is better...)